src/movie/movie.service.ts

```diff
@@ -33,26 +26,46 @@ export class MovieService {
     }
   }
 
-  async getRandomMovie(): Promise<Movie> {
+  async getRandomMovies(
+    numMovies: number = 1,
+    differentFrom: string = null,
+  ): Promise<Movie[]> {
     try {
-      // get metadata
-      const metadataUrl = `${this.METADATA_SERVICE}`
-      const metadataResponse = await axios.get(metadataUrl)
-      const metadata = Utils.extractDataFromMetadataResponse(metadataResponse)
-
-      // get poster url
-      const posterUrl = `${this.POSTER_SERVICE}?id=${metadata.imdbId}`
-      const posterResponse = await axios.get(posterUrl)
-      const poster = Utils.extractDataFromPosterResponse(posterResponse)
+      const metadata = await this.getMetadata(null, numMovies, differentFrom)
+      const posters = await Promise.all(
+        metadata.map(async (m) => await this.getPoster(m.imdbId)),
```

Comment:
Just as a hint, that should also work as `this.getPoster` already returns a Promise (you don't need to build your own):

```suggestion
        metadata.map(m => this.getPoster(m.imdbId)),
```

Reply:
Yes, I thought so too, but when I tried that it failed. However, now it works. It must have been something else. I'll adjust that accordingly.